### ISSUE TITLE
[bitnami/redis] Allow podSelector from any namespaceSelector

### ISSUE THREAD

**Issue by mwithheld** [bitnami/charts#12607]
### Name and Version

bitnami/redis 17.1.6

### What is the problem this feature will solve?

We would like an ingress netpol to allow a podSelector label from any namespace.  Using podSelector alone means the client pod must be in the same namespace, so we need to use "namespaceSelector and podSelector".
```yaml
#redis-helm-values.yaml
auth:
  enabled: false
networkPolicy:
  allowExternal: false
  enabled: true
  # Allow access if {namespace is any} and {pod has these labels}.
  ingressNSMatchLabels:
    {}
  ingressNSPodMatchLabels: 
    use_redis: v1
```

```
helm template bitnami-redis bitnami/redis -f ./redis-helm-values.yaml \
```

The problem is the network policy template does not output the namespaceSelector at all if it is set to {} (which matches any/all namespaces) -- it only outputs if you put some label in the ingressNSMatchLabels section.

```yaml
#https://github.com/bitnami/charts/tree/master/bitnami/redis)/[templates](https://github.com/bitnami/charts/tree/master/bitnami/redis/templates)/networkpolicy.yaml
...
  ingress:
...
        {{- if .Values.networkPolicy.ingressNSMatchLabels }}
        - namespaceSelector:
            matchLabels:
              {{- range $key, $value := .Values.networkPolicy.ingressNSMatchLabels }}
              {{ $key | quote }}: {{ $value | quote }}
              {{- end }}
          {{- if .Values.networkPolicy.ingressNSPodMatchLabels }}
          podSelector:
            matchLabels:
              {{- range $key, $value := .Values.networkPolicy.ingressNSPodMatchLabels }}
              {{ $key | quote }}: {{ $value | quote }}
              {{- end }}
```

### What is the feature you are proposing to solve the problem?

Update networkpolicy.yaml template to allow `ingressNSMatchLabels={}` and `podSelector=whatever:labels`

### What alternatives have you considered?

Alternative / workaround: Manual update of netpol.

**Comment by carrodher** [comment]
Thanks for reporting this issue. Would you like to contribute by creating a PR to solve the issue? The Bitnami team will be happy to review it and provide feedback. [Here](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md) you can find the contributing guidelines.

**Comment by mwithheld** [comment]
This is ready for Bitnami team review and feedback.  Thanks!

**Comment by mwithheld** [comment]
Fixed in commit 847dc4956017a5204179effd56f009ff04c9b3e4
